custom_components/google_home/api.py

```diff
@@ -197,6 +200,12 @@ async def get_alarms_and_timers(
                 ex,
             )
             device.available = False
+
+        # For some reason aiohttp returns a empty exception
+        # even if we set raise_for_status to False. (Throws exception if over 400).
+        except Exception as ex:  # pylint: disable=broad-except
+            _LOGGER.debug(ex)
```

Comment:
It's `asyncio.exceptions.TimeoutError`.
```suggestion
        except TimeoutError:
            _LOGGER.debug("Request timeout when querying %s device", device.name)
            device.available = False
```

Reply:
Please also include `device.name` in `ClientError` log message.

Reply:
No it is not `asyncio.exceptions.TimeoutError`, i just test it with changes you requested and it does not work. 

The only thing your changes does is now we express them as individual errors instead of a collective error. 

I also thought at the begning that it was `asyncio`, but found out that it is created by `aiohttp`.

Reply:
In my case it's `TimeoutError`. Ok, try this:
```python
except Exception as ex:
   _LOGGER.debug("Unknown exception for %s device: %s", device.name, type(ex))
```

Reply:
Actually it's suggested to use `asyncio.TimeoutError`. (also replace `gather` with `asyncio.gather`).

Reply:
Yes that is the error you see in the log file, but that is thrown because of an uncaught exception from `aiohttp` in `get_alarms_and_timers()`. 

`asyncio` raises `TimeoutError` if it timesout before the task is done. The task is never done because aiohttp throws an exception. The exception from `aiohttp` is just empty. I suspect this is a bug in `aiohttp`. 

If one task fails with `asyncio.gather()` it will raise `TimeoutError` and discard ALL retrieved information.

When running your proposed changes from above. I get:

```text
2021-04-10 12:44:57 DEBUG (MainThread) [custom_components.google_home] Unknown exception for Simons TV device: <class 'asyncio.exceptions.TimeoutError'>
2021-04-10 12:44:57 DEBUG (MainThread) [custom_components.google_home] Unknown exception for Køkken device: <class 'asyncio.exceptions.TimeoutError'>
```

This is expected because if `Simons TV` is the one throwing the exception, and because if one exception is thrown in `asyncio` it return that specific exceptions for all the async tasks. But when catching it broad like my initial proposal, it works.

`aiohttp` will raise an exception if you get an response status over 400. This should be disabled by default, but i don't know if HA enables it. I tried disabling it, but does not seem to mitigate the exception unfortunately.

Just to be clear I don't think that my solution is elegant or anything, but it is the only solution i have found so far.

Reply:
`asyncio.TimeoutError` is thrown by `aiohttp` on timeout. That's what you're getting and even your logs confirm that.

Reply:
I never tested with `asyncio.TimeoutError` and only `asyncio.exceptions.TimeoutError`.

It works with the former, not with the latter. I didn't see you comment before after mine was posted. 😃

Reply:
It should work with both. `asyncio` just re-exports it for better API. Probably you just had a typo. Also I noticed that incorrect logger call will be silently ignored if you have less arguments than expected by format string.

Reply:
I have experienced other "fun" problems with `aiohttp` like this before and have learned that `aiohttp` is not always consisted with there implementations. 😞